File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
       Map<Integer, Integer> frequencyByNum = new HashMap<>();
        for (int i : nums) {
            if(frequencyByNum.containsKey(i)){
                int currentFrequency = frequencyByNum.get(i);
                frequencyByNum.replace(i, currentFrequency + 1);
            } else {
            frequencyByNum.put(i, 1);
            }
        }

        return frequencyByNum.entrySet()
                .stream()
                .sorted((a, b) -> b.getValue().compareTo(a.getValue()))
                .map(entry -> entry.getKey())
                .limit(k)
                .mapToInt(i -> i)
                .toArray();
    }
}

File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-0.py
class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        self.stream = nums
        self.k = k

    def add(self, val: int) -> int:
        self.stream.append(val)
        self.stream = sorted(self.stream)
        return self.stream[-self.k]

        
